libr/anal/data.c

```diff
@@ -179,6 +183,10 @@ R_API RAnalData *r_anal_data_new_string(ut64 addr, const char *p, int len, int t
 		memcpy (ad->str, p, len);
 		ad->str[len] = 0;
 		ad->buf = malloc (len + 1);
+		if (!ad->buf) {
+			eprintf ("Cannot allocate %"PFMT64d" bytes\n", len + 1);
```

Comment:
if something goes wrong at this point, we should `free (ad->str)` and `free(ad)`.